packages/sourcegraph-extension-api/src/sourcegraph.d.ts

```diff
@@ -1245,6 +1245,30 @@ declare module 'sourcegraph' {
          * position or the current position itself.
          */
         range?: Range
+
+        /**
+         * Alerts related to this hover.
+         */
+        alerts?: Badged<HoverAlert>[]
+    }
+
+    export interface HoverAlert {
+        /**
+         * Text content to be shown on hovers.
+         */
+        summary: MarkupContent
+
+        /**
+         * When an alert has a dismissal type, dismissing it will prevent all alerts
+         * of that type from being shown. If no type is provided, the alert is not
+         * dismissible.
+         */
+        dismissalType?: string
```

Comment:
I'd suggest this naming since this name is simpler and we use string IDs for many other APIs, so the concept translates well. In the future we could reuse that ID for other things potentially, and control the dismissal with a more explicit `dismissable: boolean` property. The `type` naming is used in `DecorationType`, but there you need to manually call `createDecorationType()`, so that doesn't apply well here.
```suggestion
        id?: string
```

Reply:
I had it like that before, but it allows for `dismissible: true` but no type provided which is a bad state. I would rather 'type' than 'id' (perhaps category?), since it's not a unique ID to the alert.